security_considerations.md

```diff
@@ -10,82 +10,106 @@ if an OEM or supplier deems it necessary. In this section, we discuss several of
+
+If the Director repository is not able to update a vehicle for any reason, then it SHOULD raise the issue to the OEM.
 
 
 ## ASN.1 decoding
 
-If an OEM chooses to use ASN.1 to encode and decode metadata and other messages, then it SHOULD be careful in decoding ASN.1 messages. This is because improper decoding of ASN.1 messages may lead to arbitrary code execution or denial-of-service attacks. For example, see CVE-2016-2108 and attacks on a well-known ASN.1 compiler.
+If an OEM chooses to use ASN.1 to encode and decode metadata and other messages, then it SHOULD take great care in decoding the ASN.1 messages. Improper decoding of ASN.1 messages may lead to arbitrary code execution or denial-of-service attacks. For example, see [CVE-2016-2108](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2016-2108) and attacks on a [well-known ASN.1 compiler](https://arstechnica.com/information-technology/2016/07/software-flaw-puts-mobile-phones-and-networks-at-risk-of-complete-takeover/).
 
-In order to avoid these problems, OEMs and suppliers SHOULD use ASN.1 decoders that have been comprehensively tested via unit tests and fuzzing, whenever possible.
-Furthermore, following best practices, we recommend that the DER encoding is used instead of BER and CER, because DER provides a unique encoding of values.
+In order to avoid these problems, whenever possible OEMs and suppliers SHOULD use ASN.1 decoders that have been comprehensively tested via unit tests and fuzzing.
+
+Furthermore, following best practices, we recommend that DER encoding is used instead of BER and CER, because DER provides a unique encoding of values.
 
 
 ## Balancing EEPROM performance and security
 
-Many ECUs use EEPROM which, practically speaking, can be written to for a limited number of times. This limits how often these ECUs can be updated.
+Many ECUs use EEPROM which, practically speaking, can be written to only for a limited number of times. This in turn can impose limits on how often these ECUs can be updated.
 
 In order to analyze this problem, let us recap what new information should be downloaded in every software update cycle:
 
-1. The primary writes and sends the latest vehicle version manifest to the director repository.
-2. All secondaries write and send to the primaries fresh tokens for the time server.
-3. All ECUs download, verify, and write the latest downloaded time from the time server.
-4. All ECUs download, verify, and write metadata from the director and / or image repositories.
+1. The Primary writes and sends the latest vehicle version manifest to the Director repository.
+2. If a Time Server is used, all Secondaries write and send fresh tokens to the Primary. 
+3. All ECUs download, verify, and write the latest downloaded time from the Time Server, or whatever device is used to provide the current accurate time.
```

Comment:
Would "whatever source" be clearer than "whatever device"?

Reply:
Yes. I will fix that.